x-pack/filebeat/input/httpjson/input_test.go

```diff
@@ -26,1167 +27,1250 @@ import (
 
-	skipReason string
-}{
-	{
-		name:        "simple_GET_request",
-		setupServer: newTestServer(httptest.NewServer),
-		baseConfig: map[string]interface{}{
-			"interval":       1,
-			"request.method": http.MethodGet,
-		},
-		handler:  defaultHandler(http.MethodGet, "", ""),
-		expected: []string{`{"hello":[{"world":"moon"},{"space":[{"cake":"pumpkin"}]}]}`},
-	},
-	{
-		name:        "simple_HTTPS_GET_request",
-		setupServer: newTestServer(httptest.NewTLSServer),
-		baseConfig: map[string]interface{}{
-			"interval":                      1,
-			"request.method":                http.MethodGet,
-			"request.ssl.verification_mode": "none",
-		},
-		handler:  defaultHandler(http.MethodGet, "", ""),
-		expected: []string{`{"hello":[{"world":"moon"},{"space":[{"cake":"pumpkin"}]}]}`},
-	},
-	{
-		name:        "request_honors_rate_limit",
-		setupServer: newTestServer(httptest.NewServer),
-		baseConfig: map[string]interface{}{
-			"interval":                     1,
-			"http_method":                  http.MethodGet,
-			"request.rate_limit.limit":     `[[.last_response.header.Get "X-Rate-Limit-Limit"]]`,
-			"request.rate_limit.remaining": `[[.last_response.header.Get "X-Rate-Limit-Remaining"]]`,
-			"request.rate_limit.reset":     `[[.last_response.header.Get "X-Rate-Limit-Reset"]]`,
-		},
-		handler:  rateLimitHandler(),
-		expected: []string{`{"hello":"world"}`},
-	},
-	{
-		name:        "request_retries_when_failed",
-		setupServer: newTestServer(httptest.NewServer),
-		baseConfig: map[string]interface{}{
-			"interval":       1,
-			"request.method": http.MethodGet,
-		},
-		handler:  retryHandler(),
-		expected: []string{`{"hello":"world"}`},
-	},
-	{
-		name:        "POST_request_with_body",
-		setupServer: newTestServer(httptest.NewServer),
-		baseConfig: map[string]interface{}{
-			"interval":       1,
-			"request.method": http.MethodPost,
-			"request.body": map[string]interface{}{
-				"test": "abc",
-			},
-		},
-		handler:  defaultHandler(http.MethodPost, `{"test":"abc"}`, ""),
-		expected: []string{`{"hello":[{"world":"moon"},{"space":[{"cake":"pumpkin"}]}]}`},
-	},
-	{
-		name:        "repeated_POST_requests",
-		setupServer: newTestServer(httptest.NewServer),
-		baseConfig: map[string]interface{}{
-			"interval":       "100ms",
-			"request.method": http.MethodPost,
-		},
-		handler: defaultHandler(http.MethodPost, "", ""),
-		expected: []string{
-			`{"hello":[{"world":"moon"},{"space":[{"cake":"pumpkin"}]}]}`,
-			`{"hello":[{"world":"moon"},{"space":[{"cake":"pumpkin"}]}]}`,
-		},
-	},
-	{
-		name:        "split_by_json_objects_array",
-		setupServer: newTestServer(httptest.NewServer),
-		baseConfig: map[string]interface{}{
-			"interval":       1,
-			"request.method": http.MethodGet,
-			"response.split": map[string]interface{}{
-				"target": "body.hello",
+var (
+	expectedBulkData []string
+
+	testCases = []struct {
```

Comment:
I would prefer that this change (placing `testCases` into a `var ( )` block did not happen. It brings unnecessary diff noise at no benefit.

Also, let's make `expectedBulkData` a local var in the test function body.

Reply:
Got it. Its done.